### DISCUSSION TITLE
A module to list all orphaned config entities of core_config_data

### DISCUSSION THREAD

**Discussion by hirale** [OpenMage/magento-lts#3970] [Extensions & Modules]
Retrieve all system configurations from system.xml, compare them with the entries in the core_config_data table, and list any orphaned configurations in the backend.

https://github.com/hirale/openmage-orphaned-config

**Comment by fballiano** [comment]
nice! i'll add it to the awesome list

**Comment by empiricompany** [comment]
hi @hirale 
i've installed your module and my list of orphaned config is this
![orphaned](https://github.com/OpenMage/magento-lts/assets/5071467/f7a75547-6b51-4f65-b8f8-5ce6833901e3)


what about catalog/category/root_id? can i safe delete it?

**Reply by hirale** [reply]
no, this config was saved directly in a setup script.
![image](https://github.com/OpenMage/magento-lts/assets/22028058/0572a6b0-946b-4088-8705-4dd18131d5e7)

**Reply by empiricompany** [reply]
Ok, sorry just now I see that you had written a warning about the module and is not intended as "cleaning everything with one click" but for the selective deletion of orphaned configurations

**Reply by empiricompany** [reply]
ok it's still very useful thanks!

**Comment by hirale** [comment]
This module reads all system.xml files and retrieves configuration fields. However, configurations saved directly in PHP files cannot be retrieved, so it's advisable to create an exception list excluding such configurations before deleting any of them.

**Reply by hirale** [reply]
I have added some exclusions of core module from a fresh OpenMage v20.6.0 installation with sample data.

**Comment by sreichel** [comment]
Cool.

Can please update to follow coding-standards?
